staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -417,7 +417,7 @@ func waitForDelete(params waitForDeleteParams) ([]corev1.Pod, error) {
 		pendingPods := []corev1.Pod{}
 		for i, pod := range pods {
 			p, err := params.getPodFn(pod.Namespace, pod.Name)
-			if apierrors.IsNotFound(err) || (p != nil && p.ObjectMeta.UID != pod.ObjectMeta.UID) {
+			if apierrors.IsNotFound(err) || (err == nil && p != nil && p.ObjectMeta.UID != pod.ObjectMeta.UID) {
```

Comment:
As we discussed this in bug scrub, changes look reasonable to me. Do you think would it make sense to add a comment above this line describing the behavior?

Reply:
Good idea. Comment added.

It's actually unfortunate that it has this unnecessary `getPodFn` field because it obscures that fact that this behavior is caused by client-go.  In theory, another caller of `waitForDelete` could pass a different implementation that doesn't behave the same way.

In reality though, `waitForDeleteParams` is unexported and not used anywhere else, so it won't be different.

I think that is what this TODO comment is implying.  
https://github.com/kubernetes/kubernetes/blob/4aecb151b8301a6fc69f63c08f640c7b2b4289fc/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L249-L252

A good refactor would be to remove `getPodFn` altogether and just call client-go directly from inside this function, eliminating the unnecessary wrapper function.

Reply:
I fully agree that getPodFn is redundant and we can directly call client-go.